### PR TITLE
PMM-9150 when repo url has changed - remove dir

### PR DESCRIPTION
**Pull request by ritbl**

### CODE REVIEW

File: ci.py [hunk 1]
```diff
@@ -51,6 +52,8 @@ def merge_configs(self):
 
                 for dep in self.config['deps']:
                     if dep['name'] == override_dep['name']:
+                        if override_dep['url'] and override_dep['url'] != dep['url']:
+                            dep['repo_url_changed'] = True
```
Comment (ritbl) [inline ci.py:56]: to track that dir should be deleted

File: ci.py [hunk 2]
```diff
@@ -163,6 +173,13 @@ def create_release(self):
     def create_tags(self):
         pass
 
+    def validate_config(self):
+        for dep in self.config['deps']:
+            if not os.path.abspath(dep['path']).startswith(os.getcwd()):
```
Comment (ritbl) [inline ci.py:178]: to protect from unintended (or malicious) deletion. e.g. if `path: ../../../../sources/pmm-server` 

error message will look as follows:

```
[ERROR] 2021-11-15 11:30:21,472: For dependency [node_exporter -> /sources/node_exporter/src/github.com/prometheus/node_exporter] path must be in working directory [/home/am/dev/ritbl/prj/pmm-submodules]
```

**Review by nikita-b — APPROVED** [review]
Like :)

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-2135-fde72c8
client docker - perconalab/pmm-client-fb:PR-2135-fde72c8
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2135-fde72c8.tar.gz
Create Staging Instance: https://pmm.cd.percona.com/job/aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-2135-fde72c8&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2135-fde72c8.tar.gz

**Pull request merged**